documentation/http_api_index.markdown

```diff
@@ -0,0 +1,141 @@
+
+Using this API requires significant understanding of how Puppet's internal services are structured, but the following documents specify what is available and how to interact with it.
+
+### Configuration management services
+
+The Puppet agent application directly uses these servcies to manage the configuration of a node.
+
+These endpoints accept payload formats formatted as JSON or PSON (MIME types of `application/json` and `text/pson`, respectively) except for `File Content` and `File Bucket File`, which always use `application/octet-stream`.
+
+-   [Facts](https://puppet.com/docs/puppet/latest/http_api/http_facts.html)
+-   [Catalog](https://puppet.com/docs/puppet/latest/http_api/http_catalog.html)
+-   [Node](https://puppet.com/docs/puppet/latest/http_api/http_node.html)
+-   [File bucket file](https://puppet.com/docs/puppet/latest/http_api/http_file_bucket_file.html)
+-   [File content](https://puppet.com/docs/puppet/latest/http_api/http_file_content.html)
+-   [File metadata](https://puppet.com/docs/puppet/latest/http_api/http_file_metadata.html)
+-   [Report](https://puppet.com/docs/puppet/latest/http_api/http_report.html)
+
+### Informational services
+
+These services are not directly used by Puppet agent, but can be used by other tools.
+
+-   [Status](https://puppet.com/docs/puppet/latest/http_status.html)
+
+### Environment endpoints
+
+The `/puppet/v3/environments` and `/puppet/v3/environment/:environment` endpoints use a different format than the configuration management and informational services endpoints.
+
+These endpoints accept only payloads formatted as JSON, and respond with JSON (MIME type of `application/json`).
+
+-   [Environments](https://puppet.com/docs/puppet/latest/http_api/http_environments.html)
+-   [Environment catalog](https://puppet.com/docs/puppet/latest/http_api/http_environment.html)
+
+### Puppet Server-specific endpoints
+
+Puppet Server adds several unique endpoints of its own. They include these additional `/puppet/v3/` endpoints:
+
+-   [Environment classes](./puppet-api/v3/environment_classes.markdown), at `/puppet/v3/environment_classes`
+-   [Environment modules](./puppet-api/v3/environment_modules.markdown), at `/puppet/v3/environment_modules`
+-   [Static file content](./puppet-api/v3/static_file_content.markdown), at `/puppet/v3/static_file_content`
+
+It also includes these unique APIs, with endpoints containing other URL prefixes:
+
+-   [Status API](./status-api/v1/services.markdown), at `/status/v1/services`
+-   [Metrics v1 (mbeans) API](./metrics-api/v1/metrics_api.markdown), at `/metrics/v1/mbeans`
+-   [Metrics v2 (Jolokia) API](./metrics-api/v2/metrics_api.markdown), at `/metrics/v2/`
+-   Admin API, at `/puppet-admin-api/v1/`:
+    -   [Environment cache](./admin-api/v1/environment-cache.markdown), at `/puppet-admin-api/v1/environment-cache`
+    -   [JRuby pool](./admin-api/v1/jruby-pool.markdown), at `/puppet-admin-api/v1/jruby-pool`
+
+### Error responses
```

Comment:
Puppet Server specific, or implemented (like the CA), api endpoints have a slightly different error schema of
```
{
  "msg": "",
  "kind": ""
}
```